### PR TITLE
Require table primary key to enable basic actions

### LINKED ISSUE #31867: Cannot execute simple actions against the CSV uploaded model
### Describe the bug

Frontend blows up when trying to execute simple action against the CSV upload (automatically created model based on the CSV upload).

### To Reproduce

_Make sure "Model actions" are enabled for the Sample Database
1. Go to Admin > Settings and enable Uploads on a Sample Database with the default `PUBLIC` schema
image

2. Go to "Our analytics"
3. Click on the upload icon and upload any CSV file (you can use one from here: https://www.datablist.com/learn/csv/download-sample-csv-files)
4. Open that model actions `model/id/detail/actions`
5. Click "Crete basic actions"
6. Try to execute the "Delete" action

FE fails with "Something's gone wrong".

What's interesting is that even when you edit model's metadata and explicitly set the primary key, this still fails.

### Expected behavior

FE shouldn't result in an error page. It should rather warn that the primary key is missing (if that's the requirement) or it should assume a PK column and execute an action.

### Logs

```
utils.ts:160

Uncaught TypeError: Cannot read properties of null (reading 'forEach')
    at generateFieldSettingsFromParameters (utils.ts:160:1)
    at getOrGenerateFieldSettings (utils.ts:134:1)
    at use-action-form.ts:26:1
    at mountMemo (react-dom.development.js:15442:1)
    at Object.useMemo (react-dom.development.js:15738:1)
    at useMemo (react.development.js:1521:1)
    at useActionForm (use-action-form.ts:25:1)
    at ActionForm (ActionForm.tsx:53:1)
    at renderWithHooks (react-dom.development.js:14803:1)
    at mountIndeterminateComponent (react-dom.development.js:17482:1)


The above error occurred in the <ActionForm> component:
```

### Information about your Metabase installation

```
local dev, latest master deced97db63a583a384aceb9f44a043a54c1b697, H2, Sample Database

also on 47-RC1
```

### Severity

P1 if running actions against CSV uploads is expected behavior.

### Additional context

Trying out "Create" and "Update" actions results in 
```
Must execute implicit action on a table with a single primary key.
```

This happens before one edits model metadata (when we still have no semantic data), but also even after explicitly telling Metabase that a certain column should be PK.

### PR DESCRIPTION
**Pull request by iethree**
resolves https://github.com/metabase/metabase/issues/31867
related to, but doesn't ultimately address https://github.com/metabase/metabase/issues/31918

### Description

Implicit actions can only be executed on tables with a single primary key. This adds a check of the underlying table metadata to enable basic actions. This is a short-term fix to keep users from shooting themselves in the foot by enabling basic actions on models that cannot possibly support them.

If possible, longer-term, I think we want to support model-defined primary keys even if they differ from db-defined keys, but that's a bigger ask on the backend.

For now, this is how the UI looks

 . | Model has a pk | model does not have a pk
--- | --- | ---
**Table has a pk** | can enable basic actions | can enable basic actions
**Table does not have a pk** |  cannot enable basic actions | cannot enable basic actions

### How To test

✅ 
- Create a model based on a table with a primary key
- see that the option to enable basic actions appears in the model detail
- edit the model's metadata to not have a pk
- see that you can still enable basic actions

❌ 
- Edit a table database metadata and change the PK to an FK
- create a model based on that table
- see that you cannot enable basic actions for this model
- edit the model metadata to assign a pk to a column
- see that you still cannot enable basic actions

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase-lib/Question.ts [hunk 1]
```diff
@@ -486,7 +486,16 @@ class QuestionInner {
 
   supportsImplicitActions(): boolean {
     const query = this.query();
-    return query instanceof StructuredQuery && !query.hasAnyClauses();
+
+    // we want to check the metadata for the underlying table, not the model
+    const tableId = query.tableId();
+    const table = this.metadata().tables?.[tableId];
+    const hasSinglePk =
+      table?.fields?.filter(field => field.isPK())?.length === 1;
```
Comment (iethree) [inline frontend/src/metabase-lib/Question.ts:494]: @alxnddr Is this the right way to get at the underlying table's metadata? it works, but it feels a little dirty.

Comment (iethree) [reply frontend/src/metabase-lib/Question.ts]: I think I found a better way with `query.sourceTable()`

**Review by alxnddr — APPROVED** [review]
Looks good! I left one question, about the possibility of using MLv2 right away

File: frontend/src/metabase-lib/Question.ts [hunk 2]
```diff
@@ -486,7 +486,16 @@ class QuestionInner {
 
   supportsImplicitActions(): boolean {
     const query = this.query();
-    return query instanceof StructuredQuery && !query.hasAnyClauses();
+
+    // we want to check the metadata for the underlying table, not the model
+    const table = query.sourceTable();
+
+    const hasSinglePk =
```
[the remark this answers is not in the document]
Comment (iethree) [reply frontend/src/metabase-lib/Question.ts:493]: Good idea. I'm not sure this is fully implemented yet. the fields function always gives me an empty array.

```ts
  supportsImplicitActions(): boolean {
    const query = this.query();

    // we want to check the metadata for the underlying table, not the model
    const table = query.sourceTable();

    const queryv2 = this._getMLv2Query();
    console.log('mlV2 fields', ML.fields(queryv2, 0)); // produces []

    const hasSinglePk =
      table?.fields?.filter(field => field.isPK())?.length === 1;

    return (
      query instanceof StructuredQuery && !query.hasAnyClauses() && hasSinglePk
    );
  }
```

**Pull request merged**